src/Controller/EnMarche/SearchController.php

```diff
@@ -112,4 +114,58 @@ public function resultsAction(Request $request)
             'errors' => $errors ?? [],
         ]);
     }
+
+    /**
+     * @Route(
+     *     "/tous-les-evenements/{page}",
+     *     requirements={"page"="\d+"},
+     *     defaults={"page"=1},
+     *     name="app_event_all"
+     * )
+     * @Method("GET")
+     */
+    public function allEventsAction(int $page)
+    {
+        $eventRepository = $this->getDoctrine()->getRepository(Event::class);
+
+        $results = $eventRepository->findWithPagination($page > 1 ? SearchParametersFilter::DEFAULT_MAX_RESULTS * $page : 0);
+
+        if (!$results->count() || !($page <= (int) ceil($results->count() / SearchParametersFilter::DEFAULT_MAX_RESULTS))) {
+            throw $this->createNotFoundException();
```

Comment:
Please always add a message to exceptions to ease debugging. Here `No results found.` would be enough.